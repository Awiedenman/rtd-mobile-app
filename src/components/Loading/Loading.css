.loading-page {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100%;
  background-color: #c02d34;
}

.loading-container {
  height: 100%;
  width: 100%;
  background-color: white;
  display: flex;
  align-items: center;
  justify-content: center;
}