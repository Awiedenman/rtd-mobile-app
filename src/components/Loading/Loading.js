import React from 'react';
import './Loading.css';
import PropTypes from 'prop-types';

const LoadingPage = (props) => {
  const { type } = props;

  if (type === 'loading-page') {
    return (
      <div className='loading-page'>
        <img src='/spinner-white.svg' />
      </div>
    );
  } else if (type === 'loading-container') {
    return (
      <div className='loading-container'>
        <img src='/spinner-red.svg' />
      </div>
    );
  }

};

export default LoadingPage;

LoadingPage.propTypes = {
  type: PropTypes.string
};