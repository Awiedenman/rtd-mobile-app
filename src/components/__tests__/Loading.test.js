import React from 'react';
import { LoadingPage } from '../Loading/Loading';
import { shallow } from 'enzyme';

describe('LoadingPage', () => {
  it('should match snapShot if type == loadingPage', () => {
    const mockType='loading-page';
    const wrapper = shallow(
      <LoadingPage
        type={mockType}
      />)

      expect(wrapper).toMatchSnapshot();
  })

  it('should match snapShot if type == loadingPage', () => {
    const mockType='loading-container';
    const wrapper = shallow(
      <LoadingPage
        type={mockType}
      />)

      expect(wrapper).toMatchSnapshot();
  })
})
