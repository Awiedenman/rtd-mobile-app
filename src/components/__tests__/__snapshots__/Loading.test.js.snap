// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage should match snapShot if type == loadingPage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    type="loading-page"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="loading spinner"
        className="loading-icon-white"
        src="/spinner-white.svg"
/>,
      "className": "loading-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "loading spinner",
        "className": "loading-icon-white",
        "src": "/spinner-white.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="loading spinner"
          className="loading-icon-white"
          src="/spinner-white.svg"
/>,
        "className": "loading-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading spinner",
          "className": "loading-icon-white",
          "src": "/spinner-white.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`LoadingPage should match snapShot if type == loadingPage 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    type="loading-container"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="loading spinner"
        className="loading-icon-red"
        src="/spinner-red.svg"
/>,
      "className": "loading-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "loading spinner",
        "className": "loading-icon-red",
        "src": "/spinner-red.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="loading spinner"
          className="loading-icon-red"
          src="/spinner-red.svg"
/>,
        "className": "loading-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading spinner",
          "className": "loading-icon-red",
          "src": "/spinner-red.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
